include/common/types.h

```diff
@@ -38,6 +38,75 @@ using RemoveCVRefT = std::remove_cv_t<std::remove_reference_t<T>>;
 
 using Byte = uint8_t;
 
+namespace InterType {
+using S8 = int8_t;
+using U8 = uint8_t;
+using S16 = int16_t;
+using U16 = uint16_t;
+using S32 = int32_t;
+using U32 = uint32_t;
+using S64 = int64_t;
+using U64 = uint64_t;
+using Float32 = float ;
+using Float64 = double;
+using Char = char;
+using String = const char *;
+using Bool = bool;
+// Implementation of "record" type in interface types
+struct RecordField {
+  const char *name;
+  InterfaceType ty;
+};
+
+struct Record {
+  RecordField *field = new RecordField;
+};
+
+// Implementation of "variant" type in interface types
+struct VariantCase {
+  const char *name;
+  InterfaceType ty;
+};
+
+struct Variants {
+  VariantCase *cases = new VariantCase;
+};
+
+// Implementation of "tuple" type in interface types
+struct Tuple {
+  InterfaceType *tyTup = new InterfaceType;
+};
+
+// Implementation of "flags" type in interface types
+struct Flag {
+  char **names = new char *;
+};
+
+// Implementation of "enum" type in interface types
+struct Enum {
+  char **names = new char *;
+};
+
+// Implentation of "union" type in interface types
+struct Union {
+  InterfaceType *tyUn = new InterfaceType;
+};
+
+// Implementation of "expected" type in interface types
+struct Expecteds {
+  InterfaceType tyEx;
+  InterfaceType err;
+};
+
+/// UnknownInter definition
+struct UnknownInter {
```

Comment:
Since we have a namespace called `InterfaceType`, we can call this `Unknown` directly.